pylint/checkers/classes/class_checker.py

```diff
@@ -900,29 +900,35 @@ def _check_unused_private_functions(self, node: nodes.ClassDef) -> None:
                     n.name for n in parent_scope.nodes_of_class(nodes.Name)
                 ):
                     continue
-            for attribute in node.nodes_of_class(nodes.Attribute):
-                if (
-                    attribute.attrname != function_def.name
-                    or attribute.scope() == function_def  # We ignore recursive calls
-                ):
-                    continue
-                if isinstance(attribute.expr, nodes.Name) and attribute.expr.name in (
-                    "self",
-                    "cls",
-                    node.name,
-                ):
-                    # self.__attrname
-                    # cls.__attrname
-                    # node_name.__attrname
+            for child in node.nodes_of_class((nodes.Name, nodes.Attribute)):
+                # Check for cases where the functions are used as a variable instead of as a method call
+                if isinstance(child, nodes.Name) and child.name == function_def.name:
                     break
-                if isinstance(attribute.expr, nodes.Call):
-                    # type(self).__attrname
-                    inferred = safe_infer(attribute.expr)
+                if isinstance(child, nodes.Attribute):
+                    # Ignore recursive calls
                     if (
-                        isinstance(inferred, nodes.ClassDef)
-                        and inferred.name == node.name
+                        child.attrname != function_def.name
+                        or child.scope() == function_def
                     ):
+                        continue
+
+                    # Check self.__attrname, cls.__attrname, node_name.__attrname
+                    if isinstance(child.expr, nodes.Name) and child.expr.name in (
+                        "self",
+                        "cls",
+                        node.name,
+                    ):
```

Comment:
```suggestion
                    if isinstance(child.expr, nodes.Name) and child.expr.name in {
                        "self",
                        "cls",
                        node.name,
                    }:
```
I thought we had a check for that, "use set for membership" ?

Reply:
Perhaps that doesn't check membership checks in boolean checks? I don't know 😅